File: MoodByte/src/main/java/org/example/ModeloDTO/EntradaDTO.java
package org.example.ModeloDTO;

import org.example.Modelo.Entrada;

import java.time.LocalDate;

public record EntradaDTO(
        long id,
        String texto,
        LocalDate fechaEntrada,
        long idDiario
) {
    public EntradaDTO(Entrada entrada) {
        this(
                entrada.getId(),
                entrada.getTexto(),
                entrada.getFechaEntrada(),
                entrada.getDiario().getIdUsuario()
        );
    }

}

File: MoodByte_App/app/src/main/java/com/example/moodbyte/data/remote/dtos/EntradaDto.kt
package com.example.moodbyte.data.remote.dtos

import androidx.room.PrimaryKey
import com.example.moodbyte.data.local.entities.EntradaEntity
import com.example.moodbyte.domain.model.Entrada
import java.time.LocalDate

data class EntradaDto(
    val id: Long,
    val texto: String,
    val fechaEntrada: String,
    val idDiario: Long
)

fun EntradaDto.toEntity() = EntradaEntity(
    id = id,
    texto = texto,
    fechaEntrada = LocalDate.parse(fechaEntrada),
    idDiario = idDiario
)